### ISSUE TITLE
Multi-root workspaces: handling of nested roots (getOuterMostWorkspaceFolder)

### ISSUE THREAD

**Issue by ppiet** [rust-lang/rls-vscode#495]
TL:DR It seems that replacing workspace folders with outermost folder in `extension.ts` (`getOuterMostWorkspaceFolder(folder)`) breaks the multi-root workspace support for the case when one root is nested in another AND it's the nested one that is a Rust project. Could the use of  `getOuterMostWorkspaceFolder(folder)` be configurable? (I realize it's needed for the opposite case – when one root is nested in another and it's the outermost one that is a Rust project / has cargo.toml.)

Longer version:

I'm working in a codebase where the top-level project is not rust, but it contains rust projects in subdirs. My root folders are:

```
{
	"folders": [
		{
			"path": "~/projects/fuchsia/path/to/subproject"
		},
		{
			"path": "~/projects/fuchsia"
		}
	],
	"settings": {}
}
```
~/projects/fuchsia/path/to/subproject is a Rust project and has Cargo.toml
~/projects/fuchsia is not a Rust project and doesn't have Cargo.toml

It seems that rls-vscode almost works for this setup, but not quite – because when opening a file under subproject, we intentionally not look for the cargo file ~/projects/fuchsia/path/to/subproject , but rather in ~/projects/fuchsia , due to use of `getOuterMostWorkspaceFolder(folder)`. Could this please be configurable?

**Comment by qm3ster** [comment]
This issue also takes place when working with [Neon](https://neon-bindings.com/docs/hello-world#creating-a-new-project) projects (native nodejs modules written in Rust).
The only way to make it work correctly is opening the inner folder as a separate workspace:
```json
{
	"folders": [
		{
			"name": "Rust",
			"path": "native"
		},
		{
			"path": "."
		}
	],
	"settings": {}
}
```
